html/semantics/scripting-1/the-script-element/moving-between-documents.html

```diff
@@ -0,0 +1,139 @@
+      iframe.contentDocument.write(`<script id="s2" type="${type}" src="resources/flag-setter.js"><` + "/script>");
+      iframe.contentDocument.write("</streaming-element>");
+
+      const s = document.querySelector("#s2");
+      s.onload = t.unreached_func("onload");
+      s.onerror = t.unreached_func("onerror");
+
+      iframe.contentDocument.close();
+
+      t.step_timeout(() => {
+        assert_equals(window.didExecute, undefined, "The script must not have executed in this window");
+        assert_equals(iframe.contentWindow.didExecute, undefined,
+          "The script must not have executed in the iframe window");
+        t.done();
+      }, 3000);
+    });
+
+    document.body.appendChild(iframe);
+  }, `${type}: moving to another document during parsing, external script`);
+
+  async_test(t => {
+    t.add_cleanup(() => {
+      window.didExecute = undefined;
+    });
+
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func(() => {
+      const s = document.createElement("script");
+      s.type = type;
+      s.src = "resources/slow-flag-setter.py";
+      s.onload = t.unreached_func("onload");
+      s.onerror = t.unreached_func("onerror");
+
+      // Start the fetch
+      document.body.appendChild(s);
+
+      // Need to delay since the "prepare a script" algorithm also contains related checks; we want to
+      // test the "execute a script block" algorithm for when the fetch comes back.
+      t.step_timeout(() => {
+        iframe.contentDocument.body.appendChild(s);
+      }, 0);
+
+      t.step_timeout(() => {
+        assert_equals(window.didExecute, undefined, "The script must not have executed in this window");
+        assert_equals(iframe.contentWindow.didExecute, undefined,
+          "The script must not have executed in the iframe window");
+        t.done();
+      }, 3000);
+    });
+
+    document.body.appendChild(iframe);
+  }, `${type}: moving to another Window's document during fetching`);
+
+  async_test(t => {
+    t.add_cleanup(() => {
+      window.didExecute = undefined;
+    });
+
+    const s = document.createElement("script");
+    s.type = type;
+    s.src = "resources/slow-flag-setter.py";
+    s.onload = t.unreached_func("onload");
+    s.onerror = t.unreached_func("onerror");
+
+    // Start the fetch
+    document.body.appendChild(s);
+
+    // Need to delay since the "prepare a script" algorithm also contains related checks; we want to
+    // test the "execute a script block" algorithm for when the fetch comes back.
+    t.step_timeout(() => {
+      const doc2 = document.implementation.createHTMLDocument("title");
+      doc2.body.appendChild(s);
+    }, 0);
+
+    t.step_timeout(() => {
+      assert_equals(window.didExecute, undefined, "The script must not have executed in this window");
+      t.done();
+    }, 3000);
+  }, `${type}: moving to a document where scripting is disabled during fetching`);
```

Comment:
I think using srcdoc with sandbox might be better or good as an alternative test, since it seems likely browsers would check "browsing-context connected" and not just "connected" (and in fact this is what Chrome, Edge, and Safari seem to do).

Reply:
Hmm, on second thought, I'm unsure why we'd have such a test since it would always involve creating a new window (and thus settings object). srcdoc="" does not avoid that, as far as I can tell. So it'd be the same as other iframe tests.

If you disagree I'm happy for your to push an additional test showing me what you mean.